### PR TITLE
OMADM Framework Feature

### PR DESCRIPTION
**Pull request by dilee**

### CODE REVIEW

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheManager.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.wso2.carbon.mdm.services.android.omadm.cachemanager;
```
Comment (sinthuja) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheManager.java]: Missing license.

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheUtils.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package org.wso2.carbon.mdm.services.android.omadm.cachemanager;
```
Comment (sinthuja) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheUtils.java]: Missing license. Please add the license to all the files.

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheUtils.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import java.util.List;
+
+/**
+ * This is a utility class which contains various utilities related to the cache
+ */
+public class OMADMCacheUtils {
+
+    private static final String OMADM_CACHE_MANAGER = "OMADM_CACHE_MANAGER";
```
Comment (sinthuja) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheUtils.java]: Move the constants to the constants class if it's required by other calss as well. I just noticed that it's used here as well. OMADMCacheManagerImpl

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/impl/OMADMCacheManagerImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import javax.cache.Cache;
+
+/**
+ * This class manages all the cache related operations used in OMADM management sessions
+ */
+public class OMADMCacheManagerImpl implements OMADMCacheManager {
+
+    private static final Log log = LogFactory.getLog(OMADMCacheManagerImpl.class);
+    private static OMADMCacheManagerImpl cacheManager;
+    private static final String OMADM_CACHE_MANAGER = "OMADM_CACHE_MANAGER";
```
Comment (sinthuja) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/impl/OMADMCacheManagerImpl.java]: These attributes are not used in the class, and if so please remove it.

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheManager.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.wso2.carbon.mdm.services.android.omadm.cachemanager.beans.DMTreeOperationCacheEntry;
+
+/**
+ * The interface for the OMADM Cache Manager. The OMADMCacheManager
+ * handles all the cahce-related operations of an OMADM management session
```
Comment (warunalakshitha) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/OMADMCacheManager.java]: minor spelling mistake "cahce-related"

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/impl/OMADMCacheManagerImpl.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+package org.wso2.carbon.mdm.services.android.omadm.cachemanager.impl;
```
Comment (warunalakshitha) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/cachemanager/impl/OMADMCacheManagerImpl.java]: missing licence header

File: components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/ddf/resources/OMA-SUP-MO_DM_DevConfig.xml~ [hunk 7]
```diff
@@ -0,0 +1,249 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (warunalakshitha) [inline components/mobile-plugins/android-plugin/org.wso2.carbon.device.mgt.mobile.android.api/src/main/java/org/wso2/carbon/mdm/services/android/omadm/ddf/resources/OMA-SUP-MO_DM_DevConfig.xml~]: seems this is backup file

**Comment by charithag** [comment]
Closing this as feature is backlogged

**Pull request closed without merging**